xbmc/utils/SortUtils.cpp

```diff
@@ -179,6 +179,10 @@ std::string ByTrackNumber(SortAttribute attributes, const SortItem &values)
   return StringUtils::Format("%i", (int)values.at(FieldTrackNumber).asInteger());
 }
 
+std::string ByTotalDiscs(SortAttribute attributes, const SortItem &values)
+{
+  return StringUtils::Format("%i", (int)values.at(FieldTotalDiscs).asInteger());
```

Comment:
The only problem with that is that it doesn't then sort from largest to smallest set, it sorts by size of set _within_ an album title. Not sure that makes sense except for where I have multiple releases of the same album where the 2nd and 3rd releases have extra discs added. Figure this is to do with the label masks.

Reply:
You have me confusied by that comment. It should sort by number of discs, and _then_ by album title for those with the same number of discs, which is what you want. Didn't think there was an album label mask, just a track one.  But the sort approach (making a string and sorting that) is flawed, and that could be contributing.

Reply:
I'll have to push what I've got so far (although it's probably not very tidy at the moment!), just so you can have a look.
In a smart playlist I can get the functionality I want and if I sort the entire set with `totaldiscs > 0` I get an alphabetical list of albums with 1 disc, followed by 2 discs (again in alphabetical album order) follwed by 3 etc etc but that doesn't happen in the sideblade.  Although I added `%b` to the label formatter to get it to display the number of discs on the right (which works) it still doesn't sort on that field first.
I'll push it as soon as I can (got a busy morning!) but it's probably fairly untidy at the moment as I've been messing with around with it a lot - fair warning :)

Reply:
No rush, no pressure, I have plenty else to keep me busy

Reply:
Turns out time away was a good thing.  Fresh eyes on the code a little while ago and I realized where I had gone wrong.  Electricity board are doing some work around here tomorrow and our electric will be on and off all day apparently, so my PC is unlikely to be on much until they have finished.  In view of that, I'm going to push what I've got so far so that I have a copy on here just in case.  Ah, it's in a different branch !!  Still, gives me a little time to clean it up a bit.
It works correctly now and sorts on No. of discs -> artist -> album.  It should, in Confluence at least, display Artist - Album <some space> No.of discs in all the list views.
BTW, I set a breakpoint on `SortUtils::Sort` and was amazed at just how many times thats getting called.